src/ansys/mapdl/core/krylov.py

```diff
@@ -0,0 +1,612 @@
+
+        for iFreq in range(1, self.num_freq + 1):
+            # form RHS at the i-th frequency point
+            self.iRHS.zeros()
+            if self.load_key == 0:
+                # apply ramped loading
+                ratio = 2 * iFreq / self.num_freq
+                if self.num_freq == 1:
+                    ratio = 1.00
+            else:
+                # Apply stepped loading
+                ratio = 1.00
+
+            self.iRHS.axpy(self.fz0, ratio, 1.0)  # Get {iRHS} at iFreq
+            QtF = self.Qz.T.dot(self.iRHS)  # Reduce [Qz]^t{iRHS} -> {QtF}
+
+            # link iY to Yz(iFreq)
+            self._mapdl.vec("iY", "Z", "LINK", self.Yz.id, iFreq)
+            self.iY = self.mm.vec(name="iY")
+
+            # Reduced system at omega
+            omega = omega + self.intV * 2 * np.pi  # defined frequency value
+            ccf = -omega * omega
+
+            # modified K:[az]=[MatK]
+            self._mapdl.smat("az", "Z", "COPY", self._mat_k.id)
+            az = self.mm.mat(name="az")
+
+            # [az]=(1,0)*[az]+(ccf,0)*[MatM]
+            az.axpy(self._mat_m, ccf, 1.0)
+            # [az]=(1,0)*[az]-(0,omega)*[mat_C]
+            self._mapdl.axpy(0.0, omega, self._mat_c.id, 1.0, 0.0, az.id)
+
+            AQ = az.dot(self.Qz)  # [AQ]=[az][Qz]
+            QtAQ = self.Qz.T.dot(AQ)  # [QtAQ]=[Q^t][AQ]
+
+            # create reduced system of equations
+            s = self.mm.factorize(QtAQ)
+            # back solve for {iY}=Yz[iFreq]
+            self.iY = s.solve(QtF, self.iY)
+
+            # Output reduced solution (if requested)
+            if out_key:
+                # Yz_*.txt, reduced solution vector
+                self._mapdl.run(f"*PRINT, {self.iY.id}, {self.jobname}_Yz_{iFreq}.txt")
+        return self.Yz
+
+    def kryexpand(self, out_key=False, res_key=0):
+        """Expand reduced solution back to FE space
+
+        This method expands the reduced solution for a harmonic analysis
+        back to the original space.  Optional calculation of the residual
+        is available.
+
+        Parameters
+        ----------
+        out_key  : Bool
+                   Key to output expanded solution to Xz_*.txt file [Default:False]
+        res_key  : int, optional
+                Key to compute the residual of the expanded solution
+                = 0 means do not compute the residual [Default]
+                = 1 means compute the L-inf norm of the residual
+                = 2 means compute the L-1 norm of the residual
+                = 3 means compute the L-2 norm of the residual
+
+        Returns
+        -------
+        Ndarray
+            Solution vectors mapped to User order. (if out_key = True)
+        None
+            (if out_key = False)
+
+        Notes
+        -----
+        Distributed ANSYS Restriction: This command is not supported in
+        Distributed ANSYS."""
+
+        # Check inputs before executing the method
+        self._check_input_kryexpand(out_key, res_key)
```

Comment:
I store this parameter in the class so I can access it later in any class method.